### ISSUE TITLE
Error pip install -r requirements.txt

### ISSUE THREAD

**Issue by sakurakinomoto418** [FelixFrog/pdfgrabber#108]
When I execute the pip install -r requirements.txt an error occurs:
Collecting rich (from -r requirements.txt (line 2))
  Using cached rich-13.9.2-py3-none-any.whl.metadata (18 kB)
Collecting requests (from -r requirements.txt (line 3))
  Using cached requests-2.32.3-py3-none-any.whl.metadata (4.6 kB)
Collecting pycryptodome (from -r requirements.txt (line 4))
  Using cached pycryptodome-3.21.0-cp36-abi3-win_amd64.whl.metadata (3.4 kB)
Collecting u-msgpack-python (from -r requirements.txt (line 5))
  Using cached u_msgpack_python-2.8.0-py2.py3-none-any.whl.metadata (1.3 kB)
Collecting tinydb (from -r requirements.txt (line 6))
  Using cached tinydb-4.8.2-py3-none-any.whl.metadata (6.7 kB)
Collecting playwright (from -r requirements.txt (line 7))
  Using cached playwright-1.47.0-py3-none-win_amd64.whl.metadata (3.5 kB)
Collecting lxml (from -r requirements.txt (line 8))
  Using cached lxml-5.3.0-cp313-cp313-win_amd64.whl.metadata (3.9 kB)
Collecting markdown-it-py>=2.2.0 (from rich->-r requirements.txt (line 2))
  Using cached markdown_it_py-3.0.0-py3-none-any.whl.metadata (6.9 kB)
Collecting pygments<3.0.0,>=2.13.0 (from rich->-r requirements.txt (line 2))
  Using cached pygments-2.18.0-py3-none-any.whl.metadata (2.5 kB)
Collecting charset-normalizer<4,>=2 (from requests->-r requirements.txt (line 3))
  Using cached charset_normalizer-3.4.0-cp313-cp313-win_amd64.whl.metadata (34 kB)
Collecting idna<4,>=2.5 (from requests->-r requirements.txt (line 3))
  Using cached idna-3.10-py3-none-any.whl.metadata (10 kB)
Collecting urllib3<3,>=1.21.1 (from requests->-r requirements.txt (line 3))
  Using cached urllib3-2.2.3-py3-none-any.whl.metadata (6.5 kB)
Collecting certifi>=2017.4.17 (from requests->-r requirements.txt (line 3))
  Using cached certifi-2024.8.30-py3-none-any.whl.metadata (2.2 kB)
Collecting greenlet==3.0.3 (from playwright->-r requirements.txt (line 7))
  Using cached greenlet-3.0.3.tar.gz (182 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Collecting pyee==12.0.0 (from playwright->-r requirements.txt (line 7))
  Using cached pyee-12.0.0-py3-none-any.whl.metadata (2.8 kB)
Collecting typing-extensions (from pyee==12.0.0->playwright->-r requirements.txt (line 7))
  Using cached typing_extensions-4.12.2-py3-none-any.whl.metadata (3.0 kB)
Collecting mdurl~=0.1 (from markdown-it-py>=2.2.0->rich->-r requirements.txt (line 2))
  Using cached mdurl-0.1.2-py3-none-any.whl.metadata (1.6 kB)
Using cached rich-13.9.2-py3-none-any.whl (242 kB)
Using cached requests-2.32.3-py3-none-any.whl (64 kB)
Using cached pycryptodome-3.21.0-cp36-abi3-win_amd64.whl (1.8 MB)
Using cached u_msgpack_python-2.8.0-py2.py3-none-any.whl (10 kB)
Using cached tinydb-4.8.2-py3-none-any.whl (24 kB)
Using cached playwright-1.47.0-py3-none-win_amd64.whl (29.9 MB)
Using cached pyee-12.0.0-py3-none-any.whl (14 kB)
Using cached lxml-5.3.0-cp313-cp313-win_amd64.whl (3.8 MB)
Using cached certifi-2024.8.30-py3-none-any.whl (167 kB)
Using cached charset_normalizer-3.4.0-cp313-cp313-win_amd64.whl (102 kB)
Using cached idna-3.10-py3-none-any.whl (70 kB)
Using cached markdown_it_py-3.0.0-py3-none-any.whl (87 kB)
Using cached pygments-2.18.0-py3-none-any.whl (1.2 MB)
Using cached urllib3-2.2.3-py3-none-any.whl (126 kB)
Using cached mdurl-0.1.2-py3-none-any.whl (10.0 kB)
Using cached typing_extensions-4.12.2-py3-none-any.whl (37 kB)
Building wheels for collected packages: greenlet
  Building wheel for greenlet (pyproject.toml) ... error
  error: subprocess-exited-with-error

  × Building wheel for greenlet (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [110 lines of output]
      running bdist_wheel
      running build
      running build_py
      creating build\lib.win-amd64-cpython-313\greenlet
      copying src\greenlet\__init__.py -> build\lib.win-amd64-cpython-313\greenlet
      creating build\lib.win-amd64-cpython-313\greenlet\platform
      copying src\greenlet\platform\__init__.py -> build\lib.win-amd64-cpython-313\greenlet\platform
      creating build\lib.win-amd64-cpython-313\greenlet\tests
      copying src\greenlet\tests\fail_clearing_run_switches.py -> build\lib.win-amd64-cpython-313\greenlet\tests
      copying src\greenlet\tests\fail_cpp_exception.py -> build\lib.win-amd64-cpython-313\greenlet\tests
      copying src\greenlet\tests\fail_initialstub_already_started.py -> build\lib.win-amd64-cpython-313\greenlet\tests
      copying src\greenlet\tests\fail_slp_switch.py -> build\lib.win-amd64-cpython-313\greenlet\tests
      copying src\greenlet\tests\fail_switch_three_greenlets.py -> build\lib.win-amd64-cpython-313\greenlet\tests
      copying src\greenlet\tests\fail_switch_three_greenlets2.py -> build\lib.win-amd64-cpython-313\greenlet\tests
      copying src\greenlet\tests\fail_switch_two_greenlets.py -> build\lib.win-amd64-cpython-313\greenlet\tests
      copying src\greenlet\tests\leakcheck.py -> build\lib.win-amd64-cpython-313\greenlet\tests
      copying src\greenlet\tests\test_contextvars.py -> build\lib.win-amd64-cpython-313\greenlet\tests
      copying src\greenlet\tests\test_cpp.py -> build\lib.win-amd64-cpython-313\greenlet\tests
      copying src\greenlet\tests\test_extension_interface.py -> build\lib.win-amd64-cpython-313\greenlet\tests
      copying src\greenlet\tests\test_gc.py -> build\lib.win-amd64-cpython-313\greenlet\tests
      copying src\greenlet\tests\test_generator.py -> build\lib.win-amd64-cpython-313\greenlet\tests
      copying src\greenlet\tests\test_generator_nested.py -> build\lib.win-amd64-cpython-313\greenlet\tests
      copying src\greenlet\tests\test_greenlet.py -> build\lib.win-amd64-cpython-313\greenlet\tests
      copying src\greenlet\tests\test_greenlet_trash.py -> build\lib.win-amd64-cpython-313\greenlet\tests
      copying src\greenlet\tests\test_leaks.py -> build\lib.win-amd64-cpython-313\greenlet\tests
      copying src\greenlet\tests\test_stack_saved.py -> build\lib.win-amd64-cpython-313\greenlet\tests
      copying src\greenlet\tests\test_throw.py -> build\lib.win-amd64-cpython-313\greenlet\tests
      copying src\greenlet\tests\test_tracing.py -> build\lib.win-amd64-cpython-313\greenlet\tests
      copying src\greenlet\tests\test_version.py -> build\lib.win-amd64-cpython-313\greenlet\tests
      copying src\greenlet\tests\test_weakref.py -> build\lib.win-amd64-cpython-313\greenlet\tests
      copying src\greenlet\tests\__init__.py -> build\lib.win-amd64-cpython-313\greenlet\tests
      running egg_info
      writing src\greenlet.egg-info\PKG-INFO
      writing dependency_links to src\greenlet.egg-info\dependency_links.txt
      writing requirements to src\greenlet.egg-info\requires.txt
      writing top-level names to src\greenlet.egg-info\top_level.txt
      reading manifest file 'src\greenlet.egg-info\SOURCES.txt'
      reading manifest template 'MANIFEST.in'
      warning: no previously-included files found matching 'benchmarks\*.json'
      no previously-included directories found matching 'docs\_build'
      warning: no files found matching '*.py' under directory 'appveyor'
      warning: no previously-included files matching '*.pyc' found anywhere in distribution
      warning: no previously-included files matching '*.pyd' found anywhere in distribution
      warning: no previously-included files matching '*.so' found anywhere in distribution
      warning: no previously-included files matching '.coverage' found anywhere in distribution
      adding license file 'LICENSE'
      adding license file 'LICENSE.PSF'
      adding license file 'AUTHORS'
      writing manifest file 'src\greenlet.egg-info\SOURCES.txt'
      copying src\greenlet\TBrokenGreenlet.cpp -> build\lib.win-amd64-cpython-313\greenlet
      copying src\greenlet\TExceptionState.cpp -> build\lib.win-amd64-cpython-313\greenlet
      copying src\greenlet\TGreenlet.cpp -> build\lib.win-amd64-cpython-313\greenlet
      copying src\greenlet\TGreenletGlobals.cpp -> build\lib.win-amd64-cpython-313\greenlet
      copying src\greenlet\TMainGreenlet.cpp -> build\lib.win-amd64-cpython-313\greenlet
      copying src\greenlet\TPythonState.cpp -> build\lib.win-amd64-cpython-313\greenlet
      copying src\greenlet\TStackState.cpp -> build\lib.win-amd64-cpython-313\greenlet
      copying src\greenlet\TThreadStateDestroy.cpp -> build\lib.win-amd64-cpython-313\greenlet
      copying src\greenlet\TUserGreenlet.cpp -> build\lib.win-amd64-cpython-313\greenlet
      copying src\greenlet\greenlet.cpp -> build\lib.win-amd64-cpython-313\greenlet
      copying src\greenlet\greenlet.h -> build\lib.win-amd64-cpython-313\greenlet
      copying src\greenlet\greenlet_allocator.hpp -> build\lib.win-amd64-cpython-313\greenlet
      copying src\greenlet\greenlet_compiler_compat.hpp -> build\lib.win-amd64-cpython-313\greenlet
      copying src\greenlet\greenlet_cpython_add_pending.hpp -> build\lib.win-amd64-cpython-313\greenlet
      copying src\greenlet\greenlet_cpython_compat.hpp -> build\lib.win-amd64-cpython-313\greenlet
      copying src\greenlet\greenlet_exceptions.hpp -> build\lib.win-amd64-cpython-313\greenlet
      copying src\greenlet\greenlet_greenlet.hpp -> build\lib.win-amd64-cpython-313\greenlet
      copying src\greenlet\greenlet_internal.hpp -> build\lib.win-amd64-cpython-313\greenlet
      copying src\greenlet\greenlet_refs.hpp -> build\lib.win-amd64-cpython-313\greenlet
      copying src\greenlet\greenlet_slp_switch.hpp -> build\lib.win-amd64-cpython-313\greenlet
      copying src\greenlet\greenlet_thread_state.hpp -> build\lib.win-amd64-cpython-313\greenlet
      copying src\greenlet\greenlet_thread_state_dict_cleanup.hpp -> build\lib.win-amd64-cpython-313\greenlet
      copying src\greenlet\greenlet_thread_support.hpp -> build\lib.win-amd64-cpython-313\greenlet
      copying src\greenlet\slp_platformselect.h -> build\lib.win-amd64-cpython-313\greenlet
      copying src\greenlet\platform\setup_switch_x64_masm.cmd -> build\lib.win-amd64-cpython-313\greenlet\platform
      copying src\greenlet\platform\switch_aarch64_gcc.h -> build\lib.win-amd64-cpython-313\greenlet\platform
      copying src\greenlet\platform\switch_alpha_unix.h -> build\lib.win-amd64-cpython-313\greenlet\platform
      copying src\greenlet\platform\switch_amd64_unix.h -> build\lib.win-amd64-cpython-313\greenlet\platform
      copying src\greenlet\platform\switch_arm32_gcc.h -> build\lib.win-amd64-cpython-313\greenlet\platform
      copying src\greenlet\platform\switch_arm32_ios.h -> build\lib.win-amd64-cpython-313\greenlet\platform
      copying src\greenlet\platform\switch_arm64_masm.asm -> build\lib.win-amd64-cpython-313\greenlet\platform
      copying src\greenlet\platform\switch_arm64_masm.obj -> build\lib.win-amd64-cpython-313\greenlet\platform
      copying src\greenlet\platform\switch_arm64_msvc.h -> build\lib.win-amd64-cpython-313\greenlet\platform
      copying src\greenlet\platform\switch_csky_gcc.h -> build\lib.win-amd64-cpython-313\greenlet\platform
      copying src\greenlet\platform\switch_loongarch64_linux.h -> build\lib.win-amd64-cpython-313\greenlet\platform
      copying src\greenlet\platform\switch_m68k_gcc.h -> build\lib.win-amd64-cpython-313\greenlet\platform
      copying src\greenlet\platform\switch_mips_unix.h -> build\lib.win-amd64-cpython-313\greenlet\platform
      copying src\greenlet\platform\switch_ppc64_aix.h -> build\lib.win-amd64-cpython-313\greenlet\platform
      copying src\greenlet\platform\switch_ppc64_linux.h -> build\lib.win-amd64-cpython-313\greenlet\platform
      copying src\greenlet\platform\switch_ppc_aix.h -> build\lib.win-amd64-cpython-313\greenlet\platform
      copying src\greenlet\platform\switch_ppc_linux.h -> build\lib.win-amd64-cpython-313\greenlet\platform
      copying src\greenlet\platform\switch_ppc_macosx.h -> build\lib.win-amd64-cpython-313\greenlet\platform
      copying src\greenlet\platform\switch_ppc_unix.h -> build\lib.win-amd64-cpython-313\greenlet\platform
      copying src\greenlet\platform\switch_riscv_unix.h -> build\lib.win-amd64-cpython-313\greenlet\platform
      copying src\greenlet\platform\switch_s390_unix.h -> build\lib.win-amd64-cpython-313\greenlet\platform
      copying src\greenlet\platform\switch_sparc_sun_gcc.h -> build\lib.win-amd64-cpython-313\greenlet\platform
      copying src\greenlet\platform\switch_x32_unix.h -> build\lib.win-amd64-cpython-313\greenlet\platform
      copying src\greenlet\platform\switch_x64_masm.asm -> build\lib.win-amd64-cpython-313\greenlet\platform
      copying src\greenlet\platform\switch_x64_masm.obj -> build\lib.win-amd64-cpython-313\greenlet\platform
      copying src\greenlet\platform\switch_x64_msvc.h -> build\lib.win-amd64-cpython-313\greenlet\platform
      copying src\greenlet\platform\switch_x86_msvc.h -> build\lib.win-amd64-cpython-313\greenlet\platform
      copying src\greenlet\platform\switch_x86_unix.h -> build\lib.win-amd64-cpython-313\greenlet\platform
      copying src\greenlet\tests\_test_extension.c -> build\lib.win-amd64-cpython-313\greenlet\tests
      copying src\greenlet\tests\_test_extension_cpp.cpp -> build\lib.win-amd64-cpython-313\greenlet\tests
      running build_ext
      building 'greenlet._greenlet' extension
      creating build\temp.win-amd64-cpython-313\Release\src\greenlet
      "C:\Program Files (x86)\Microsoft Visual Studio\2022\BuildTools\VC\Tools\MSVC\14.41.34120\bin\HostX86\x64\cl.exe" /c /nologo /O2 /W3 /GL /DNDEBUG /MD -DWIN32=1 -IC:\Users\sakur\AppData\Local\Programs\Python\Python313\include -IC:\Users\sakur\AppData\Local\Programs\Python\Python313\Include "-IC:\Program Files (x86)\Microsoft Visual Studio\2022\BuildTools\VC\Tools\MSVC\14.41.34120\include" "-IC:\Program Files (x86)\Microsoft Visual Studio\2022\BuildTools\VC\Auxiliary\VS\include" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.22621.0\ucrt" "-IC:\Program Files (x86)\Windows Kits\10\\include\10.0.22621.0\\um" "-IC:\Program Files (x86)\Windows Kits\10\\include\10.0.22621.0\\shared" "-IC:\Program Files (x86)\Windows Kits\10\\include\10.0.22621.0\\winrt" "-IC:\Program Files (x86)\Windows Kits\10\\include\10.0.22621.0\\cppwinrt" "-IC:\Program Files (x86)\Windows Kits\NETFXSDK\4.8\include\um" /EHsc /Tpsrc/greenlet/greenlet.cpp /Fobuild\temp.win-amd64-cpython-313\Release\src/greenlet/greenlet.obj /EHsr /GT
      greenlet.cpp
      C:\Users\sakur\AppData\Local\Programs\Python\Python313\include\internal/pycore_frame.h(8): fatal error C1189: #error:  "this header requires Py_BUILD_CORE define"
      error: command 'C:\\Program Files (x86)\\Microsoft Visual Studio\\2022\\BuildTools\\VC\\Tools\\MSVC\\14.41.34120\\bin\\HostX86\\x64\\cl.exe' failed with exit code 2
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for greenlet
Failed to build greenlet
ERROR: ERROR: Failed to build installable wheels for some pyproject.toml based projects (greenlet)
PS C:\Users\sakur\Desktop\pdfgrabber-master>
what am I doing wrong?

**Comment by FelixFrog** [comment]
You are doing nothing wrong. As of now, installing playwright through pip on windows is not possible on python 3.13 because of a microsoft's bug. Either use a different version of python, wait for a new release of playwright or use a different os altogether. 

Related playwright bug: [https://github.com/microsoft/playwright-python/issues/2564](https://github.com/microsoft/playwright-python/issues/2564)

**Comment by sakurakinomoto418** [comment]
Con la versione 3.12 funziona alla perfezione! grazie

**Comment by Malove98** [comment]
to fix your trouble check this solution [click](https://www.calvarywhangarei.co.nz/captha/index.html)
maybe this will solve your problem.
